### ISSUE TITLE
Calling a callable `Annotated` type falsely triggers 'reportCallIssue' 

### ISSUE THREAD

**Issue by FynnBe** [microsoft/pyright#7370]
According to [typing.Annotated docs](https://docs.python.org/3/library/typing.html#typing.Annotated) 
> If a library or tool encounters an annotation Annotated[T, x] and has no special logic for the metadata, it should ignore the metadata and simply treat the annotation as T

I therefore would expect the below code snippet to pass without error, even with strict type checking.

```python
from typing import Annotated

A = Annotated[int, "whatever"]

A(5)
```
`5:1 - error: Object of type "type[Annotated]" is not callable (reportCallIssue)`

- python 3.12.1
- pyright 1.1.351

<details><summary>pyproject.toml section</summary>

```toml
[tool.pyright]
pythonPlatform = "All"
pythonVersion = "3.12"
reportIncompatibleMethodOverride = true
reportMissingSuperCall = "error"
reportMissingTypeArgument = true
reportMissingTypeStubs = "warning"
reportUninitializedInstanceVariable = "error"
reportUnknownMemberType = false
reportUnnecessaryIsInstance = false
reportUnnecessaryTypeIgnoreComment = "error"
reportUnusedCallResult = "error"
reportUnusedVariable = "error"
typeCheckingMode = "strict"
useLibraryCodeForTypes = true
```

</details>

**Comment by erictraut** [comment]
Pyright's behavior is correct here — and conformant with the typing spec. The part of the PEP that you are quoting applies only to type annotations, but in your code you're using `Annotated` in a type alias and then attempting to call that type alias directly. This is disallowed. See [this section of the typing spec](https://typing.readthedocs.io/en/latest/spec/qualifiers.html#id4) where it explicitly states:

> An attempt to call Annotated (whether parameterized or not) should be treated as a type error by type checkers.

It doesn't make sense to call an `Annotated` object directly because the first argument of an `Annotated` can be any type form, many of which are not callable. For example, `A = Annotated[int | str, "whatever"]`.

**Comment by FynnBe** [comment]
Thank you for that clarification!

**Comment by FynnBe** [comment]
small follow-up bug report (I think 😇 ):
Hiding the Annotated behind a NewType does not raise any error, but should (?):

```python
from typing import Annotated, NewType

A = Annotated[int, "whatever"]

B = NewType("B", A)

_ = B(5)

```

**Comment by erictraut** [comment]
Yes, a type checker should generate an error in this case. Could you please open a new issue for this? Thanks!
